src/3rd_party_repos.c

```diff
@@ -266,9 +266,14 @@ enum swupd_code third_party_set_repo(const char *state_dir, const char *path_pre
 	set_content_url(repo->url);
 	set_version_url(repo->url);
 
-	/* set up swupd to use the certificate from the 3rd-party repository */
+	/* set up swupd to use the certificate from the 3rd-party repository,
+	 * unless the user is specifying a path for the certificate to use */
 	string_or_die(&repo_cert_path, "%s/%s/%s/%s", path_prefix, SWUPD_3RD_PARTY_BUNDLES_DIR, repo->name, CERT_PATH);
-	set_cert_path(repo_cert_path);
+	if (strcmp(globals.cert_path, CERT_PATH) == 0) {
```

Comment:
What if you run:
```swupd 3rd-party bundle-add test -C /usr/share/clear/update-ca/Swupd_Root.pem```

Reply:
then that value would be overwritten. The problem is that we don't have a way of telling if the user set up a global variable or not... so this approach works fine with the exception of the one corner case you just mentioned. Do you have another suggestion that would not have this limitation?

Reply:
It's not a minor corner case. If you have a 3rd party repo that is maintained by the same maintainer as your main distro you may want to use the same public key to validate that, so I think that this is a very possible use case.
I'll take a look at the code and see what we can do about that.